src/hist/basehist.py

```diff
@@ -417,6 +417,14 @@ def plot1d(
         cat_ax = self.axes[overlay]
         cats = cat_ax if cat_ax.traits.discrete else np.arange(len(cat_ax.centers))
         d1hists = [self[{overlay: cat}] for cat in cats]
+        if "label" in kwargs:
+            if isinstance(kwargs["label"], list) and len(kwargs["label"]) == len(cats):
```

Comment:
Does it _have_ to be a list? Couldn't it be any arbitrary collection? Except a string. I assume you are trying not to match on a string?